src/runtime/logger.js

```diff
@@ -20,14 +20,20 @@ function getLogger (params) {
 
   const transports = [
     new winston.transports.Console({ format }),
-    new winston.transports.File({ filename: 'stdout', format })
   ]
 
-  if (params.testOutputsPath) {
+  if (process.title !== 'browser') {
     transports.push(new winston.transports.File({
       format,
-      filename: path.join(params.testOutputsPath, 'run.out')
+      filename: 'stdout'
     }))
+
+    if (!params.testOutputsPath) {
+      transports.push(new winston.transports.File({
+        format,
+        filename: path.join(params.testOutputsPath, 'run.out')
+      }))
+    }
   }
 
   return winston.createLogger({
```

Comment:
(creating a thread)

You listed a lot of `fallback` libraries, do you know what purpose they solve? (for example the path library might "just" be required by the winston logger).

@achingbrain shared an interesting option: It might be enough to use the `browser` field to shim the features that are not isomorphic (the loggers creation and the env retrieval): https://github.com/defunctzombie/package-browser-field-spec#replace-specific-files---advanced

A few examples he shared:
```js
// get-env.js
export function getEnv() {
  return process.env
}

// get-env.browser.js
export function getEnv {
  return window.testground.env
}

// package.json
"exports": { // used by external imports
  "./get-env": {
    "browser": "get-env.browser.js",
    "import": "get-env.js"
  }
},
"browser": { // used by internal imports (e.g by esbuild before building code for running unit tests)
  "get-env.js": "get-env.browser.js"
}
```

Reply:
Oh ok nice, so kinda like what you would do with C macro's in C/C++, go file suffixes and  cfg-like macro's in Rust. Got-cha. That does look a lot cleaner indeed. Will play with it this week and see how far I get.

Reply:
Ok read the spec, seems fine. Going to adapt the PR to this format.